wsd/DocumentBroker.cpp

```diff
@@ -1726,13 +1726,7 @@ void DocumentBroker::handleUploadToStorageResponse(const StorageBase::UploadResu
     _unitWsd.onDocumentUploaded(lastUploadSuccessful);
 
 #if !MOBILEAPP
-    if (lastUploadSuccessful && !isModified())
-    {
-        // Flag the document as un-modified in the admin console.
-        // But only when we have uploaded successfully and the document
-        // is current not flagged as modified by Core.
-        Admin::instance().modificationAlert(_docKey, getPid(), false);
-    }
+    Admin::instance().uploadedAlert(_docKey, getPid(), lastUploadSuccessful);
```

Comment:
I think this should be `lastUploadSuccessful && !isModified()`.

Because if by the time we finish uploading there is further modification to the document (i.e. `isModified() == true`), we will set uploaded here to `true`, which is wrong (since there is a newer version to be uploaded). And since the `DocumentBroker::setModified` is already called with `true` argument, we will have the uploaded state wrong.

Instead, if here we use `lastUploadSuccessful && !isModified()`, then in that case the uploaded state will be correct and when the user does save and upload, it will get updated here when and if by the time we upload the document is not marked as modified.

However, there is a race condition when the `modifiedstatus=false` arrives late from Core. For that case, see my notes in `DocumentBroker::setModified()`.

Reply:
thanks